packages/app/src/EstimatorCacheGoogleCloudStorage.ts

```diff
@@ -33,6 +27,7 @@ export default class EstimatorCacheGoogleCloudStorage
     grouping: string,
   ): Promise<void> {
     const cachedEstimates = await this.getCloudFileContent(grouping)
+    const bucketName = configLoader().GCP.CACHE_BUCKET_NAME
```

Comment:
I'm not quite following why refactoring the bucket name out of the constructor is needed for your use case. I'm also not sure it's an improvement... I think I find doing one call to get the config and pass it into the constructor easier to understand/maintain - but I might be missing something 🙃

Reply:
Again, same reply as above. The CacheManager was being instantiated at the time the code was being executed, which in turn was instantiated an EstimatorCacheGoogleCLoudStorage. Moving this out of the constructor means that it doesn't request the config until it's needed and allows the config to be set first.